pandas/core/arrays/timedeltas.py

```diff
@@ -997,6 +989,30 @@ def objects_to_td64ns(data, unit="ns", errors="raise"):
     return result.view('timedelta64[ns]')
 
 
+def _validate_td64_dtype(dtype):
+    try:
+        if dtype == np.dtype("timedelta64"):
+            dtype = _TD_DTYPE
+            msg = textwrap.dedent("""\
+                Passing in 'timedelta' dtype with no precision is deprecated
+                and will raise in a future version. Please pass in
+                'timedelta64[ns]' instead.""")
+            warnings.warn(msg, FutureWarning, stacklevel=4)
+    except TypeError:
+        # extension dtype
+        pass
+
+    try:
```

Comment:
you can avoid all of this try/except/else if you use ``not is_dtype_equal(dtype, _TD_DTYPE)